### PR TITLE
feat: Add Academic Year start Sept-Aug[2.42-DHIS2-15803]

### PR DESCRIPTION
**Pull request by d-bernat**
To address the unique needs of academic institutions, we propose including the academic calendar as an option within our analytics apps. This enhancement will allow institutions to align their data with the academic year, which typically runs from September to August.

Currently, we support "academic" years as fixed periods (e.g., 2024Sep). Please note, we will not introduce a new relative period type for academic years. Introducing financial years as a separate type was not ideal; a more streamlined approach would have been to use a single yearly type, such as "Yearly (start January)" for the standard year and "Yearly (start October)" for financial years.

### CODE REVIEW

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/event-query.json [hunk 1]
```diff
@@ -13,6 +13,13 @@
       "version": {
         "min": 39
       }
+    },
+    {
+      "name": "AcademicYear2023Sep",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/event-query.json:18]:
```suggestion
      "name": "financialYear2023Sep",
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/event-aggregated.json [hunk 2]
```diff
@@ -1,4 +1,11 @@
 {
   "scenarios": [
+    {
+      "name": "AcademicYear2023Sep",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/event-aggregated.json:4]:
```suggestion
      "name": "financialYear2023Sep",
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/tracked-entity-query.json [hunk 3]
```diff
@@ -125,6 +125,13 @@
       "version": {
         "min": 41
       }
+    },
+    {
+      "name": "AcademicYear2018Sep",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/tracked-entity-query.json:130]:
```suggestion
      "name": "financialYear2018Sep",
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/enroll-query.json [hunk 4]
```diff
@@ -6,6 +6,13 @@
       "version": {
         "min": 39
       }
+    },
+    {
+      "name": "AcademicYear2022Sep",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/enroll-query.json:11]:
```suggestion
      "name": "financialYear2022Sep",
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/enroll-aggregated.json [hunk 5]
```diff
@@ -1,4 +1,11 @@
 {
   "scenarios": [
+    {
+      "name": "AcademicYear2022Sep",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/enroll-aggregated.json:4]:
```suggestion
      "name": "financialYear2022Sep",
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json [hunk 6]
```diff
@@ -41,6 +41,13 @@
       "version": {
         "min": 42
       }
+    },
+    {
+      "name": "AcademicYear2022Sep",
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/generator/scenarios/aggregated.json:46]:
```suggestion
      "name": "financialYear2022Sep",
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/event/query/EventsQuery4AutoTest.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+/** Groups e2e tests for "/events/query" endpoint. */
+public class EventsQuery4AutoTest extends AnalyticsApiTest {
+  private final AnalyticsEventActions actions = new AnalyticsEventActions();
+
+  @Test
+  public void AcademicYear2023Sep() throws JSONException {
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/event/query/EventsQuery4AutoTest.java:49]:
```suggestion
  public void financialYear2023Sep() throws JSONException {
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/event/aggregate/EventsAggregate9AutoTest.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+/** Groups e2e tests for "/events/aggregate" endpoint. */
+public class EventsAggregate9AutoTest extends AnalyticsApiTest {
+  private final AnalyticsEventActions actions = new AnalyticsEventActions();
+
+  @Test
+  public void AcademicYear2023Sep() throws JSONException {
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/event/aggregate/EventsAggregate9AutoTest.java:49]:
```suggestion
  public void financialYear2023Sep() throws JSONException {
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/enrollment/query/EnrollmentsQuery5AutoTest.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+/** Groups e2e tests for "/enrollments/query" endpoint. */
+public class EnrollmentsQuery5AutoTest extends AnalyticsApiTest {
+  private final AnalyticsEnrollmentsActions actions = new AnalyticsEnrollmentsActions();
+
+  @Test
+  public void AcademicYear2022Sep() throws JSONException {
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/enrollment/query/EnrollmentsQuery5AutoTest.java:49]:
```suggestion
  public void financialYear2022Sep() throws JSONException {
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/enrollment/aggregate/EnrollmentsAggregate6AutoTest.java [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+/** Groups e2e tests for "/enrollments/aggregate" endpoint. */
+public class EnrollmentsAggregate6AutoTest extends AnalyticsApiTest {
+  private final AnalyticsEnrollmentsActions actions = new AnalyticsEnrollmentsActions();
+
+  @Test
+  public void AcademicYear2022Sep() throws JSONException {
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/enrollment/aggregate/EnrollmentsAggregate6AutoTest.java:49]:
```suggestion
  public void financialYear2022Sep() throws JSONException {
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/aggregate/AnalyticsQueryDv12AutoTest.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+public class AnalyticsQueryDv12AutoTest extends AnalyticsApiTest {
+  private RestApiActions actions;
+
+  @BeforeAll
+  public void setup() {
+    actions = new RestApiActions("analytics");
+  }
+
+  @Test
+  public void AcademicYear2022Sep() throws JSONException {
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/aggregate/AnalyticsQueryDv12AutoTest.java:57]:
```suggestion
  public void financialYear2022Sep() throws JSONException {
```

File: dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/trackedentity/TrackedEntityQuery8AutoTest.java [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+/** Groups e2e tests for "/trackedEntities/query" endpoint. */
+public class TrackedEntityQuery8AutoTest extends AnalyticsApiTest {
+  private AnalyticsTrackedEntityActions actions = new AnalyticsTrackedEntityActions();
+
+  @Test
+  public void AcademicYear2018Sep() throws JSONException {
```
Comment (maikelarabori) [inline dhis-2/dhis-test-e2e/src/test/java/org/hisp/dhis/analytics/trackedentity/TrackedEntityQuery8AutoTest.java:51]:
```suggestion
  public void financialYear2018Sep() throws JSONException {
```

**Review by maikelarabori — APPROVED** [review]
Looks good Dusan, just minor suggestions. Please take a look.

Thx!

**Review by gnespolino — APPROVED** [review]

**Pull request merged**